src/docfx/lib/markdown/MarkdownEngine.cs

```diff
@@ -298,7 +298,16 @@ private bool IsArchived(FilePath path)
         private (string? content, object? file) ReadFile(string path, MarkdownObject origin)
         {
             var status = t_status.Value!.Peek();
-            var (error, file) = _linkResolver.ResolveContent(new SourceInfo<string>(path, origin.GetSourceInfo()), origin.GetFilePath());
+            var contentFallback = true;
+            var referencingFile = origin.GetFilePath();
+            if (referencingFile.Path.Value.EndsWith(".yml"))
+            {
+                contentFallback = _templateEngine.GetSchema(_documentProvider.GetMime(referencingFile)).ContentFallback ?? contentFallback;
```

Comment:
This approach doesn't seem to have the right layering, SDP (`JsonSchemaTransformer`) is a layer atop markdown, expose `contentFallback` as a markup option and pass it down from `JsonSchemaTrasnformer`.

Reply:
Ok, move to JsonSchemaTransformer and pass along contentFallback 

---
In reply to: [489133378](https://github.com/dotnet/docfx/pull/6558#discussion_r489133378) [](ancestors = 489133378)